payments-ui-core/src/main/java/com/stripe/android/ui/core/elements/PostalCodeConfig.kt

```diff
@@ -37,17 +37,17 @@ internal class PostalCodeConfig(
 
         override fun getError(): FieldError? {
             return when {
-                input.isNotBlank() && !isFull() && country == "US" -> {
+                input.isNotBlank() && !isValid() && country == "US" -> {
                     FieldError(R.string.address_zip_invalid)
                 }
-                input.isNotBlank() && !isFull() -> {
+                input.isNotBlank() && !isValid() -> {
                     FieldError(R.string.address_zip_postal_invalid)
                 }
                 else -> null
             }
         }
 
-        override fun isFull(): Boolean = input.length >= format.minimumLength
+        override fun isFull(): Boolean = input.length >= format.maximumLength
```

Comment:
While this doesn’t impact the resolution of the issue, it makes more sense to me than the previous implementation: A field is full when it reaches the maximum length, not when it exceeds the minimum length.